tools/niminst/niminst.nim

```diff
@@ -749,22 +845,107 @@ proc archiveDist(c: var ConfigData) =
   let oldDir = getCurrentDir()
   setCurrentDir(tmpDir)
   try:
+    # Timestamp to set the files to. This make sure that the created archive
+    # is deterministic with regards to creation time.
+    let timestamp =
+      try:
+        parse(c.commitdate, "yyyy-MM-dd", utc()).toTime()
+      except TimeParseError:
+        # If no time is provided, use epoch 0
+        fromUnix(0)
+
+    const
+      AvoidFragments = [
+        DirSep & ".DS_Store", # macOS directory metadata
+        DirSep & "__MACOSX"   # macOS resource forks
+      ]
+        ## Path fragments to avoid packaging
+
+      DefaultPermission = {fpUserRead, fpUserWrite, fpGroupRead, fpOthersRead}
+        ## Permissions applied to files
+      
+      ExecutablePermission = DefaultPermission + {fpUserExec, fpGroupExec, fpOthersExec}
+        ## Permissions applied to executables/directories
+
+    for path in walkDirRec(proj, {pcFile, pcDir}, checkDir = true):
+      if AvoidFragments.allIt(it notin path):
+        # Normalize permissions so that a reasonably stricter/laxer umask won't
+        # cause undeterminism
+        if fpUserExec in getFilePermissions(path):
```

Comment:
I would prefer to invert this such that only the part of the code that varies is covered by the if, while the rest remains as is.